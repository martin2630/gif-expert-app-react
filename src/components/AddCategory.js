import React, {useState} from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
    // const [{setCategories}] = this.props;
    const initialInputValue = ''
    const [inputValue, setInputValue] = useState(initialInputValue);

    const handleInputChange = (e) => {
        setInputValue(e.target.value)
        console.log(" handleInputChange llamado");
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length>2){
            setCategories(cats => [inputValue, ...cats])
            setInputValue(initialInputValue)
        }
        return 
    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{ inputValue }</p>
            <input 
                type="text" 
                onChange={handleInputChange} 
                value={inputValue} 
                placeholder="Rick y Morty"
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
