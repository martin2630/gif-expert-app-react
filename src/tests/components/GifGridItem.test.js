import { React } from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe("Pruebas en <GifGridItem />", () => {
    const title = "Hola mundo";
    const url = "https://holamundo.com";
    const wrapper = shallow(<GifGridItem title={ title } url={ url }/>)

    test('debe de mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar un texto en el tÃ­tulo', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    })

    test('debe de tener una img igual a url y alt de los props', () => {
        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    })

    test('debe de tener la clase animated__fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect( className.includes('animated__fadeIn') ).toBe( true );
    })
    
})