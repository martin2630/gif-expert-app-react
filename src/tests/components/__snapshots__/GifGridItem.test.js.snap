// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animated__fadeIn"
>
  <img
    alt="Hola mundo"
    src="https://holamundo.com"
  />
  <p>
    Hola mundo
  </p>
</div>
`;
